modules/programs/bat.nix

```diff
@@ -0,0 +1,43 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.bat;
+
+in
+
+{
+  meta.maintainers = [ maintainers.marsam ];
+
+  options.programs.bat = {
+    enable = mkEnableOption "Bat";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.bat;
+      defaultText = "pkgs.bat";
+      description = ''
+        Bat package to install.
+      '';
+    };
+
+    config = mkOption {
+      type = types.lines;
```

Comment:
Perhaps this could be of type `types.attrsOf types.str`? I.e. to have something like
```nix
xdg.configFile."bat/config" = mkIf (cfg.config != {}) {
  text = concatStringsSep "\n" (
    mapAttrsToList (n: v: ''--${n}="${v}"'') cfg.config
  );
};
```